public/app/features/alerting/unified/api/prometheus.ts

```diff
@@ -45,17 +54,46 @@ export function prepareRulesFilterQueryParams(
   return Object.fromEntries(params);
 }
 
-export async function fetchRules(dataSourceName: string, filter?: FetchPromRulesFilter): Promise<RuleNamespace[]> {
+export function paramsWithMatcherAndState(
+  params: Record<string, string | string[]>,
+  state?: string[],
+  matchers?: Matcher[]
+) {
+  let paramsResult = { ...params };
+
+  if (state?.length) {
+    paramsResult = { ...paramsResult, state };
+  }
+
+  if (matchers?.length) {
+    const matcherToJsonString: string[] = matchers.map((m) => `${JSON.stringify(m)}`);
```

Comment:
Why are we wrapping `JSON.stringify` in a string literal here?

Reply:
removing it!